auth-setup.ts

```diff
@@ -0,0 +1,364 @@
+      message: "Would you like to enable magic link email authentication?",
+    },
+    {
+      type: "confirm",
+      name: "enableWebAuthn",
+      message: "Would you like to enable WebAuthn authentication?",
+    },
+    {
+      type: "confirm",
+      name: "enableSmtp",
+      message: "Would you like to enable SMTP?",
+    },
+  ];
+
+  const answers = await inquirer.prompt(questions);
+
+  const providersDetails: [string, any][] = [];
+  for (const provider of answers.providers) {
+    const existingProvider = existingConfig.providers.find(
+      (p) => p.name === provider,
+    );
+    const providerDetails = await inquirer.prompt([
+      {
+        type: "input",
+        name: "clientId",
+        message: `Enter the ${provider} client ID:`,
+        default: existingProvider?.client_id,
+      },
+      {
+        type: "input",
+        name: "secret",
+        message: `Enter the ${provider} secret:`,
+        default: existingProvider?.secret,
+      },
+    ]);
+    providersDetails.push([provider, providerDetails]);
+  }
+
+  let query = `
+    CONFIGURE CURRENT DATABASE
+    RESET ext::auth::ProviderConfig;
+
+    CONFIGURE CURRENT DATABASE
+    RESET ext::auth::AuthConfig;
+
+    CONFIGURE CURRENT DATABASE
+    RESET ext::auth::UIConfig;
+
+    CONFIGURE CURRENT DATABASE
+    RESET ext::auth::SMTPConfig;
+
+    CONFIGURE CURRENT DATABASE SET
+    ext::auth::AuthConfig::auth_signing_key := '${answers.authSigningKey}';
+  `;
+
+  if (answers.tokenTTL) {
+    query += `
+      CONFIGURE CURRENT DATABASE SET
+      ext::auth::AuthConfig::token_time_to_live := <duration>'${answers.tokenTTL}';
+    `;
+  }
+
+  const PROVIDER_MAP: Record<string, string> = {
+    github: "GitHubOAuthProvider",
+    google: "GoogleOAuthProvider",
+    apple: "AppleOAuthProvider",
+    azure: "AzureOAuthProvider",
+    discord: "DiscordOAuthProvider",
+    slack: "SlackOAuthProvider",
+  };
+
+  for (const [provider, providerDetails] of providersDetails) {
+    const providerType = PROVIDER_MAP[provider];
+    query += `
+      CONFIGURE CURRENT DATABASE
+      INSERT ext::auth::${providerType} {
+        secret := '${providerDetails.secret}',
+        client_id := '${providerDetails.clientId}'
+      };
+    `;
+  }
+
+  if (answers.enablePasswordAuth) {
+    const passwordAuthConfig = await inquirer.prompt([
+      {
+        type: "confirm",
+        name: "requireVerification",
+        message: "Should email/password require email verification?",
+      },
+    ]);
+    query += `
+      CONFIGURE CURRENT DATABASE
+      INSERT ext::auth::EmailPasswordProviderConfig {
+        require_verification := <bool>${passwordAuthConfig.requireVerification}
+      };
+    `;
+  }
+
+  if (answers.enableMagicLink) {
+    const magicLinkAuthConfig = await inquirer.prompt([
+      {
+        type: "input",
+        name: "tokenTTL",
+        message:
+          "How long should the email sign-in link be valid for? (ISO Duration format)?",
```

Comment:
We should probably put the default here:

```suggestion
        message:
          "How long should the email sign-in link be valid for? (ISO Duration format)?",
        default: "PT10M",
```